src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1531,6 +1615,122 @@ public void MediaType_SetThenGet_ValuesMatch(Uri remoteServer)
             Assert.Equal(MediaType, request.MediaType);
         }
 
+        [Theory, MemberData(nameof(MixedWebRequestParameters))]
+        public void GetResponseAsync_ParametersAreNotCachable_CreateNewClient(HttpWebRequestParameters requestParameters, bool connectionReusedParameter)
+        {
+            var handle = RemoteExecutor.Invoke(async (serializedParameters, connectionReusedString) =>
+            {
+                var parameters = JsonSerializer.Deserialize<HttpWebRequestParameters>(serializedParameters);
+
+                using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                {
+                    listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                    listener.Listen(1);
+                    var ep = (IPEndPoint)listener.LocalEndPoint;
+                    var uri = new Uri($"http://{ep.Address}:{ep.Port}/");
+
+                    HttpWebRequest request0 = WebRequest.CreateHttp(uri);
+                    HttpWebRequest request1 = WebRequest.CreateHttp(uri);
+                    parameters.Configure(request0);
+                    parameters.Configure(request1);
+                    request0.Method = HttpMethod.Get.Method;
+                    request1.Method = HttpMethod.Get.Method;
+
+                    string responseContent = "Test response.";
+
+                    Task<WebResponse> firstResponseTask = request0.GetResponseAsync();
+                    using (Socket server = await listener.AcceptAsync())
+                    using (var serverStream = new NetworkStream(server, ownsSocket: false))
+                    using (var serverReader = new StreamReader(serverStream))
+                    {
+                        await ReplyToClient(responseContent, server, serverReader);
+                        await VerifyResponse(responseContent, firstResponseTask);
+
+                        Task<Socket> secondAccept = listener.AcceptAsync();
+
+                        Task<WebResponse> secondResponseTask = request1.GetResponseAsync();
+                        await ReplyToClient(responseContent, server, serverReader);
+                        if (bool.Parse(connectionReusedString))
+                        {
+                            Assert.False(secondAccept.IsCompleted);
+                            await VerifyResponse(responseContent, secondResponseTask);
+                        }
+                        else
+                        {
+                            await VerifyNewConnection(responseContent, secondAccept, secondResponseTask);
+                        }
+                    }
+                }
+                return RemoteExecutor.SuccessExitCode;
+            }, JsonSerializer.Serialize<HttpWebRequestParameters>(requestParameters), connectionReusedParameter.ToString());
+
+            Assert.Equal(RemoteExecutor.SuccessExitCode, handle.ExitCode);
```

Comment:
You don't need to store the handle or add this assert manually... just do `RemoveInvoke.Execute(...).Dispose()`, and the Dispose will do the appropriate checks.  In fact, I'd expect the tests to fail non-deterministically due to not using Dispose, as there's logic in the handle to validate that Dispose is being called.